src/cli/rustup_mode.rs

```diff
@@ -1472,3 +1541,9 @@ fn output_completion_script(shell: Shell, command: CompletionCommand) -> Result<
 
     Ok(())
 }
+
+fn utc_from_manifest_date(date_str: &str) -> Option<Date<Utc>> {
+    NaiveDate::parse_from_str(date_str, "%Y-%m-%d")
+        .ok()
+        .map(|date| Utc.from_utc_date(&date))
+}
```

Comment:
This function looks like it is a direct copy of code from elsewhere which might be better lifted into a utility function if it's not already in one.